Implemented update status for admin to change request status

okikiola-jkt opened the pull request.

okikiola-jkt committed:
Implemented update status for admin to change request status

fob413 commented on src/controllers/admin/changeStatus.js:
```diff
@@ -0,0 +1,24 @@
+    const {id} = request.params;
+
+    try{
+        const query = 'UPDATE requests SET status = $1 WHERE id =$2';
```
You should first of all check if the request exists, before attempting to update it.

fob413 commented on src/routes.js:
```diff
@@ -44,7 +45,8 @@ app.delete('/request/:id', apiAuth, deleteUserRequest);
 app.get('/admin/request/:id', adminApiAuth, getSingleRequest);
-app.get('/admin/users', adminApiAuth, getAllUsers)
+app.get('/admin/users', adminApiAuth, getAllUsers);
+app.put('/admin/request/:id',adminApiAuth, changeStatus);
```
You need a middleware to validate the request body

fob413 reviewed (changes requested):
Nice one.. Just 2 small tiny comments

okikiola-jkt committed:
Implemented feedback
Added check status middleware to check for correct status
Implemented validate request exists

fob413 commented on src/middleware/admin/checkStatus.js:
```diff
@@ -0,0 +1,23 @@
+  
+});
+
+const ChangeStatusMiddleware = (request, response, next) => {
```
you naming convention here is wrong. Variables starting with capital letters are meant for classes, not regular functions.

Change this to `changeStatusMiddleware`

okikiola-jkt committed:
Implemented feedback

fob413 commented on src/controllers/admin/changeStatus.js:
```diff
@@ -0,0 +1,39 @@
+        });
+    }
+
+    try{
```
This should cover the entire code

okikiola-jkt committed:
Implemented feedback

okikiola-jkt committed:
Minor changes

okikiola-jkt committed:
Minor change
